### PR TITLE
Consisten itergp naming

### LINKED ISSUE #4: Create a proper package
Currently the repo is in a very ad hoc state. We want to package it up to make it usable for others. 

Tasks:
- Package properly with a UUID
- Move code to src/IterGP.jl and other src/*.jl
- Add tests for different strategies not crashing and converging
- Move example scripts to examples/* with custom envs

### PR DESCRIPTION
**Pull request by SebastianCallh**
Closes https://github.com/SebastianCallh/IterGP.jl/issues/4

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,10 +52,10 @@ scatter!(plt, x, y, label="Data", color=2)
 
 
 # Implementation details
-The user facing API exposes three primitives: the `IterGP` constructor, various policites, e.g. `ConjugateGradientPolicy`, and the `CholeskyPreconditioner`. Having used these primitives to construct a `GP`, you can call `AbstractGPs.posterior` on it, which is where the magic happens.
+The IterGP API builds around `AbstractGPs.GP`, and exposes policies which can be used to call `IterGPs.posterior`, which is where the magic happens.
```
Comment (sethaxen) [inline README.md]:
```suggestion
The IterGP API builds around `AbstractGPs.AbstractGP`, and exposes policies that can be used to call `IterGP.posterior`, which is where the magic happens.
```

**Review by sethaxen — COMMENTED** [review]
There are several more instances of "IterGPs" still in the Readme that I can't propose changes to. Also one in `test/runtests.jl`

**Comment by SebastianCallh** [comment]
Thanks! That was sloppy of me

**Review by sethaxen — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**